### PR TITLE
fix: test wrapper did only work on master branch

### PR DESCRIPTION
**Pull request by makkes**
**What type of PR is this?**

Bug

**What this PR does/ why we need it**:

When trying to merge PRs into another branch (e.g. release/3.2), the
test-wrapper would still check the master branch. Now the script is
configurable using command-line arguments:

```
go run -tags experimental scripts/test-wrapper origin release/3.2
```

The first parameter is the remote's name and the second parameter is
the target branch name.

In addition to merging this commit, a change to this TeamCity build
configuration is necessary to pass these two parameters to the script:

https://teamcity.mesosphere.io/buildConfiguration/kubeaddons_KubernetesBaseAddons?mode=builds

**Which issue(s) this PR fixes**:

no issue

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: test/go.mod [hunk 1]
```diff
@@ -13,10 +13,17 @@ require (
 	github.com/datawire/ambassador v1.11.0
 	github.com/docker/docker v1.4.2-0.20200203170920-46ec8731fbce
 	github.com/google/go-cmp v0.5.4
+<<<<<<< Updated upstream
```
Comment (hectorj2f) [inline test/go.mod:16]: conflicts here

**Review by armandgrillet — APPROVED** [review]
LGTM. Will we be able to change the TC step to use the branch where the run is happening in the command so that we don't need to manually set the branch there?

**Comment by makkes** [comment]
> LGTM. Will we be able to change the TC step to use the branch where the run is happening in the command so that we don't need to manually set the branch there?

I was assuming that the TC job will be configurable in such a way that it dynamically passes the correct parameters, yess.

File: test/scripts/test-wrapper.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ import (
 	testutils "github.com/mesosphere/kubeaddons/test/utils"
 )
 
-const (
+var (
 	upstreamRemote = "origin"
 	upstreamBranch = "master"
 )
```
Comment (jimmidyson) [inline test/scripts/test-wrapper.go:24]: nit: can you make these consts again and call them `defaultUpstreamRemote` etc (mutable globals are bleugh - yeah i know this is just a small utility but still...)

```suggestion
const (
	defaultupstreamRemote = "origin"
	defaultUpstreamBranch = "master"
)
```

File: test/scripts/test-wrapper.go [hunk 3]
```diff
@@ -28,6 +28,12 @@ type addonName string
 var re = regexp.MustCompile(`^addons/([a-zA-Z-]+)/?`)
 
 func main() {
+	if len(os.Args) > 1 {
+		upstreamRemote = os.Args[1]
+	}
+	if len(os.Args) > 2 {
+		upstreamBranch = os.Args[2]
+	}
```
Comment (jimmidyson) [inline test/scripts/test-wrapper.go:36]:
```suggestion
	upstreamRemote := defaultUpstreamRemote
	if len(os.Args) > 1 {
		upstreamRemote = os.Args[1]
	}
	upstreamBranch := defaultUpstreamBranch
	if len(os.Args) > 2 {
		upstreamBranch = os.Args[2]
	}
```

**Review by jimmidyson — APPROVED** [review]
a nit but functionality looks good

**Review by joejulian — APPROVED** [review]

**Comment by joejulian** [comment]
@mesosphere-mergebot backport release/3.3

**Pull request merged**